resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -182,10 +185,25 @@ private[spark] class KubernetesClusterSchedulerBackend(
       } else if (totalExpectedExecutors.get() <= runningExecutorPods.size) {
         logDebug("Maximum allowed executor limit reached. Not scaling up further.")
       } else {
+        val executorPodsWithIPs = EXECUTOR_PODS_BY_IPS_LOCK.synchronized {
+            executorPodsByIPs.values.toList
```

Comment:
does this make a defensive copy of the values?  I think we want a defensive copy here